src/components/PostsList.tsx

```diff
@@ -1,56 +1,56 @@
 import classNames from 'classnames';
 import React from 'react';
-import { Post } from '../types/Post';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from '../app/hooks';
+import { setSelectedPost } from '../features/selectedPost/selectedPost';
 
-type Props = {
-  posts: Post[],
-  selectedPostId?: number,
-  onPostSelected: (post: Post | null) => void,
-};
-
-export const PostsList: React.FC<Props> = ({
-  posts,
-  selectedPostId = 0,
-  onPostSelected,
-}) => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+export const PostsList: React.FC = () => {
+  const { items } = useAppSelector(state => state.posts);
```

Comment:
Consider using createSelector from '@reduxjs/toolkit' to extract derived data from Redux state.